### PR TITLE
MCOL-5385 This patch reduces RAM consumption and adds GROUP_CONCAT RA…

### PR DESCRIPTION
**Pull request by drrtuy**
…M accounting feature

### CODE REVIEW

File: dbcon/joblist/groupconcat.cpp [hunk 1]
```diff
@@ -1038,17 +1064,18 @@ void GroupConcatNoOrder::getResult(uint8_t* buff, const string& sep)
       outputRow(oss, fRow);
       fRow.nextRow();
     }
-
+    size_t sizeDiff = oss.str().size() - prevResultSize;
```
Comment (mariadb-DenisKhalikov) [inline dbcon/joblist/groupconcat.cpp:1067]: Could it be a situation when prevResultSize > oss.str().size(), if so we could face possible signed integer overflow.

Comment (drrtuy) [reply dbcon/joblist/groupconcat.cpp:1067]: I don't see how it is possible given that oss size is a monotonic non-decreasing function? Could you give an example?

Comment (mariadb-DenisKhalikov) [reply dbcon/joblist/groupconcat.cpp:1067]: Good to know, never heard about it.

**Review by mariadb-DenisKhalikov — APPROVED** [review]

File: dbcon/joblist/groupconcat.cpp (hunk 1 above)
Comment (drrtuy) [reply dbcon/joblist/groupconcat.cpp:1067]: I mean every iteration of a while loop either adds or doesn't change oss so sizeDiff can't be negative.

**Pull request merged**